apps/explorer/src/components/OwnedObjectsV2/OwnedObjects.tsx

```diff
@@ -40,32 +33,32 @@ function OwnerCoins({ id }: { id: string }): JSX.Element {
                     <Heading color="gray-90" variant="heading4/semibold">
                         NFTs
                     </Heading>
-
-                    <div className="flex max-h-80 flex-col overflow-auto">
-                        <div className="flex flex-wrap">
-                            {results
-                                .slice(
-                                    (currentSlice - 1) * OBJECTS_PER_PAGE,
-                                    currentSlice * OBJECTS_PER_PAGE
-                                )
-                                .map((obj) => (
+                    {isLoading || isFetching ? (
+                        <LoadingSpinner />
+                    ) : (
+                        <div className="flex max-h-80 flex-col overflow-auto">
+                            <div className="flex flex-wrap">
+                                {data.pages[currentPage].data.map((obj) => (
```

Comment:
Is this safe here? Could we run into issues with `data.pages[currentPage]` being `undefined`

Reply:
I believe the required value is to output an empty array based on the docs but i added an optional here just in case